### DISCUSSION TITLE
[Flutter] How to detect if the app is running in Maestro?

### DISCUSSION THREAD

**Discussion by shrijanRegmi** [mobile-dev-inc/Maestro#1739] [Q&A]
I saw an example in Kotlin to detect if the app is running in Maestro. I want the flutter equivalent. How do I do that?

**Comment by portyu9** [comment]
Hi @shrijanRegmi 

The current recommended way is to explicitly pass a launch argument from Maestro.

```yaml
- launchApp:
    arguments:
      isMaestro: "true"
```

For Flutter, you can retrieve launch arguments with a package such as `flutter_launch_arguments`:

```dart
import 'package:flutter_launch_arguments/flutter_launch_arguments.dart';

final fla = FlutterLaunchArguments();
final isMaestro = await fla.getString('isMaestro') == 'true';
```

Then use `isMaestro` for whatever test-only configuration you need.

This is preferable to detecting Maestro by checking ports such as `7001`/`22087`. Maestro now documents that port-based approach as deprecated, unsupported in Maestro Cloud, and potentially removable in the future.

Please mark this answer as accepted if it helped, thank you.
